idaes/generic_models/properties/core/coolprop/coolprop_forms.py

```diff
@@ -0,0 +1,281 @@
+    Returns:
+        Pyomo expression matching CoolProp exponential sum form
+    """
+    # y = yc * exp(Tc/T * sum(ni*theta^ti))
+    Tc = cobj.temperature_crit
+    theta = 1 - T/Tc
+
+    s = _nt_sum(cobj, prop, theta)
+
+    return yc*exp(s)
+
+
+def expression_exponential_tau(cobj, prop, T, yc):
+    """
+    Method for creating expressions for CoolProp exponential sum forms with tau
+
+    Args:
+        cobj - component object that will contian the parameters
+        prop - name of property parameters are associated with
+        T - temperature to use in expression
+        yc - value of property at critical point
+
+    Returns:
+        Pyomo expression matching CoolProp exponential sum form with tau
+    """
+    # y = yc * exp(Tc/T * sum(ni*theta^ti))
+    Tc = cobj.temperature_crit
+    theta = 1 - T/Tc
+
+    s = _nt_sum(cobj, prop, theta)
+
+    return yc*exp(Tc/T*s)
+
+
+def dT_expression_exponential_tau(cobj, prop, T, yc):
+    """
+    Method for creating expressions for temeprature derivative of CoolProp
+    exponential sum forms with tau
+
+    Args:
+        cobj - component object that will contian the parameters
+        prop - name of property parameters are associated with
+        T - temperature to use in expression
+        yc - value of property at critical point
+
+    Returns:
+        Pyomo expression for temeprature derivative of CoolProp exponential sum
+        form with tau
+    """
+    # y = yc * exp(Tc/T * sum(ni*theta^ti))
+    # Need d(y)/dT
+
+    y = expression_exponential_tau(cobj, prop, T, yc)
+
+    Tc = cobj.temperature_crit
+
+    sdT = _nt_sum_dT(cobj, prop, T, Tc)
+
+    return y*sdT
+
+
+def expression_nonexponential(cobj, prop, T, yc):
+    """
+    Method for creating expressions for CoolProp non-exponential sum forms
+
+    Args:
+        cobj - component object that will contian the parameters
+        prop - name of property parameters are associated with
+        T - temperature to use in expression
+        yc - value of property at critical point
+
+    Returns:
+        Pyomo expression mathcing CoolProp non-exponential sum form
+    """
+    # y = yc * (1 + sum(ni*theta^ti))
+    Tc = cobj.temperature_crit
+    theta = 1 - T/Tc
+
+    s = _nt_sum(cobj, prop, theta)
+
+    return yc*(1+s)
+
+
+def parameters_polynomial(cobj, prop, prop_units, alist, blist):
+    """
+    Method for creating parameters for expression forms using A-B parameters
+    (rational polynomial forms)
+
+    Args:
+        cobj - component object that will contian the parameters
+        prop - name of property parameters are associated with
+        prop_units - units of measurment for property
+        Alist - list of values for A-parameter
+        Blist - list of values for B-parameter
+
+    Returns:
+        None
+    """
+    for i in range(0, len(alist)):
+        aval = alist[i]
+        if i == 0:
+            param_units = prop_units
+        else:
+            param_units = prop_units/pyunits.K**i
```

Comment:
On the whole, this doesn't worry me too much. Yes, it would be nice if we could have different units for each `VarData` to have different units in this case, I don not think it is worth a performance/memory hit. Further, allowing different units for each `VarData` just opens the door to more confusion of the naive user.